spark/src/test/java/org/apache/iceberg/spark/source/TestDataSourceOptions.java

```diff
@@ -135,33 +134,35 @@ public void testNoWriteFormatOption() throws IOException {
   public void testHadoopOptions() throws IOException {
     String tableLocation = temp.newFolder("iceberg-table").toString();
 
-    Configuration customHadoopConf = new Configuration(CONF);
-    customHadoopConf.set(ConfigProperties.COMPRESS_METADATA, "true");
-
-    HadoopTables tables = new HadoopTables(customHadoopConf);
-    PartitionSpec spec = PartitionSpec.unpartitioned();
-    Map<String, String> options = Maps.newHashMap();
-    tables.create(SCHEMA, spec, options, tableLocation);
-
-    List<SimpleRecord> expectedRecords = Lists.newArrayList(
-        new SimpleRecord(1, "a"),
-        new SimpleRecord(2, "b")
-    );
-    Dataset<Row> originalDf = spark.createDataFrame(expectedRecords, SimpleRecord.class);
-    originalDf.select("id", "data").write()
-        .format("iceberg")
-        .mode("append")
-        .option("hadoop.iceberg.compress.metadata", "true")
-        .save(tableLocation);
-
-    Dataset<Row> resultDf = spark.read()
-        .format("iceberg")
-        .option("hadoop.iceberg.compress.metadata", "true")
-        .load(tableLocation);
-    List<SimpleRecord> resultRecords = resultDf.orderBy("id")
-        .as(Encoders.bean(SimpleRecord.class))
-        .collectAsList();
-
-    Assert.assertEquals("Records should match", expectedRecords, resultRecords);
+    try {
+      spark.sparkContext().hadoopConfiguration().set("fs.default.name", "hdfs://localhost:9000");
```

Comment:
Could you add some comments to give this more context? Is this testing that tables work if you create metadata using one FS prefix and read/write with another?

Reply:
Yeah, I'll add a comment. Basically, this test verifies that `hadoop.` data source options are propagated correctly. Otherwise, it would fail as there is no HDFS.

Reply:
The table is created with another Hadoop config. I'll move this line after the table creation.